src/OpenTelemetry.Api/Internal/SelfDiagnostics.cs

```diff
@@ -0,0 +1,72 @@
+﻿// <copyright file="SelfDiagnostics.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Internal
+{
+    /// <summary>
+    /// Self diagnostics class captures the EventSource events sent by OpenTelemetry
+    /// modules and writes them to local file for troubleshooting.
+    /// </summary>
+    public class SelfDiagnostics : IDisposable
+    {
+        /// <summary>
+        /// Long-living object that hold relevant resources, including an EventListener.
+        /// </summary>
+        private static readonly SelfDiagnostics Instance = new SelfDiagnostics();
```

Comment:
Is this intended to be a singleton? If yes, why would the ctor be public?

Reply:
I've updated it.